Get position of GZip.Stream in compressed bytes

quinnj opened the pull request:
This PR changes `position` to `position(::GZip.Stream,of_raw::Bool=false)`, with `position(f,true)` returning the value of `gzoffset`.

My use case is splitting gzipped files into smaller chunks by entire lines, so I open a gzip file, read a line, then write it out to a new GZip.Stream. In the mean time, I need to track how many compressed bytes my new gzip file is to know when to close it.

kmsquire commented on src/GZip.jl:
```diff
@@ -208,12 +208,12 @@ let _zlib_h = dlopen(_zlib)
         const _gzseek = :gzseek
         const _gztell = :gztell
-        #_gzoffset = :gzoffset    ## not implemented
+        const _gzoffset = :gzoffset    ## not implemented
```
I would remove the `## not implemented` comment here and above.

kmsquire commented:
I think you need to rebase this off of master.

quinnj commented:
Yeah, I got the commits a little mixed up. Should be fixed now.

quinnj commented:
Ok, think I finally got it figured out.

kmsquire commented on src/GZip.jl:
```diff
@@ -340,8 +340,9 @@ skip(s::GZipStream, n::Integer) =
 
-position(s::GZipStream) =
-    ccall((_gztell, _zlib), ZFileOffset, (Ptr{Void},), s.gz_file)
+position(s::GZipStream, of_raw::Bool=false) = of_raw ?
```
Minor nit: I would simply call this `raw`.  But since it's not really exposed anywhere, I only care enough to write this comment. ;-)

kmsquire commented on src/GZip.jl:
```diff
@@ -219,12 +219,12 @@ let _zlib_h = Libdl.dlopen(_zlib)
         const _gzseek = :gzseek64
         const _gztell = :gztell64
-        #_gzoffset = :gzoffset64  ## not implemented
+        const _gzoffset = :gzoffset64
```
It's quite possible this wasn't implemented because it didn't exist in earlier versions of zlib.  At the least, the version on travis doesn't seem to find it.

kmsquire commented:
When I run this locally, I get:

```julia
julia> Pkg.test("GZip")
INFO: Testing GZip
5765
5765
ERROR: LoadError: test failed: (1775 == 5765)
 in expression: position(gzfile,true) == pos
 in error at error.jl:21
 in default_handler at test.jl:29
 in do_test at test.jl:52
 in include at ./boot.jl:252
 in include_from_node1 at loading.jl:134
 in process_options at ./client.jl:310
 in _start at ./client.jl:409
while loading /Users/kevin/.julia/v0.4/GZip/test/runtests.jl, in expression starting on line 96
```

quinnj commented:
As far as I can tell, it was version 1.2.3 where they re-orged a lot of the repository and introduced `gzoffset`; that was back in 2011.

kmsquire commented:
(Ignore the `5670`'s--I had added some debug print statments.  Removed inline on github.)

kmsquire commented:
Maybe there's no `gzoffset64` then?  At least in the version on github?

https://travis-ci.org/JuliaLang/GZip.jl/jobs/61714711#L129

quinnj commented:
Ok, fixed the test, renamed the arg to `raw` and added `GZLIB_VERSION` and made calling `gzoffset` conditional on having a recent enough version of the library. Hopefully that will do it?

quinnj commented:
Weird, not sure why it's not finding `gzoffset64`. I'm seeing that defined all the way back to when `gzoffset` was first introduced.

quinnj commented:
Good to merge?

kmsquire commented:
Would you be willing to change all of the versions from strings to Versions?  I don't think the Version type existed when this was originally written, and Versions would make more sense when testing, well, versions.

quinnj commented:
The problem is that `VersionNumber` doesn't support `1.2.3.9`, only `1.2.3`. I figured it was better to be more accurate vs. truncate the version number.

kmsquire commented:
Ah, right, forgot about that limitation. I proposed once to extend Version (for BinDeps/Debian versioning), and that wasn't accepted, but it would be great to have extended version functionality in a package so we could use it for these things.

quinnj commented:
Yeah, seems like that would be good to have. Thanks for merging. I'll tag a version in METADATA.
